plugins/hypervisors/vmware/src/main/java/com/cloud/storage/resource/VmwareStorageProcessor.java

```diff
@@ -2683,15 +2683,14 @@ public Answer deleteVolume(DeleteCommand cmd) {
                         List<VirtualDisk> virtualDisks = vmMo.getVirtualDisks();
                         List<String> managedDatastoreNames = getManagedDatastoreNamesFromVirtualDisks(virtualDisks);
 
+                        // Preserve other disks of the VM
```

Comment:
Thanks for these changes @shwstppr. 
Can you please test destroy VM and expunge VM cases with VM having disks. Because this particular block is removed to address disk detachments in case of destroy VM. If those operations are not effected, this looks good to me.

Reply:
@harikrishna-patnala I've verified the destroy case and have seen any issue.
Will try to share API responses if needed